### PR TITLE
build/deploy: do not deploy streamlink-latest, and remove old nightlies

### PR DESCRIPTION
**Pull request by beardypig**
This updates the Travis/BinTray deploy to stop deploying `streamlink-latest.exe` and to delete old nightly versions so we don't run out of quota on BinTray.

### CODE REVIEW

File: script/bintrayupdate.sh [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
-curl -u "${BINTRAY_USER}:${BINTRAY_KEY}" -H "Content-Type: application/json" -d @"${build_dir}/bintray-changelog.json" ${latest_api_url}
+api_base="https://api.bintray.com/packages/streamlink/streamlink-nightly/streamlink"
+current_api_url="${api_base}/versions/$(date +'%Y.%m.%d')/release_notes"
+files_api_url="${api_base}/files"
+versions_api_url="${api_base}/versions"
+
+curl -u "${BINTRAY_USER}:${BINTRAY_KEY}" -H "Content-Type: application/json" -d @"${build_dir}/bintray-changelog.json" "${current_api_url}"
+
+echo "Deleting versions older than ${old_date}..."
+versions=$(curl -s -u "${BINTRAY_USER}:${BINTRAY_KEY}" "${files_api_url}" | grep -Po '(?<="version":")[^"]+(?=")')
```
Comment (bastimeyer) [inline script/bintrayupdate.sh]: Does TravisCI have `jq` installed by default? Then you could replace the (unsafe) regex
`grep -Po '(?<="version":")[^"]+(?=")'`
with a simple
`jq -r '.[].version'`
that returns the version property of all objects in the json array response in a raw format (no quotations)

Comment (beardypig) [reply script/bintrayupdate.sh]: I doubt it, but if there is an ubuntu package it could be added... lemme see

Comment (bastimeyer) [reply script/bintrayupdate.sh]: Looks like `jq` needs to be added to the apt package list in the travisci config. Package is whitelisted, just checked it.

Comment (beardypig) [reply script/bintrayupdate.sh]: Cool. I read that it's pre-installed, can test it out.

File: script/bintrayupdate.sh [hunk 2]
```diff
@@ -9,10 +9,10 @@ current_api_url="${api_base}/versions/$(date +'%Y.%m.%d')/release_notes"
 files_api_url="${api_base}/files"
 versions_api_url="${api_base}/versions"
 
-curl -u "${BINTRAY_USER}:${BINTRAY_KEY}" -H "Content-Type: application/json" -d @"${build_dir}/bintray-changelog.json" "${current_api_url}"
+#curl -u "${BINTRAY_USER}:${BINTRAY_KEY}" -H "Content-Type: application/json" -d @"${build_dir}/bintray-changelog.json" "${current_api_url}"
```
[the remark this answers is not in the document]
Comment (beardypig) [reply script/bintrayupdate.sh]: damn... that was for testing :(

File: script/bintrayupdate.sh (hunk 1 above)
Comment (beardypig) [reply script/bintrayupdate.sh]: `jq` is cool, thanks for the tip

**Comment by gravyboat** [comment]
Good stuff, thanks @beardypig.

**Pull request merged**